libraries/AP_Motors/AP_MotorsMatrix.h

```diff
@@ -86,6 +86,19 @@ class AP_MotorsMatrix : public AP_MotorsMulticopter {
     // add_motor using raw roll, pitch, throttle and yaw factors
     void                add_motor_raw(int8_t motor_num, float roll_fac, float pitch_fac, float yaw_fac, uint8_t testing_order, float throttle_factor = 1.0f);
 
+    // structure to statically store motor information.  Entries match
+    // the arguments to add_motor - with the exception of the type,
+    // for compactness.
+    struct MotorDefInt {
+        int8_t motor_num;
```

Comment:
Removing this saves you 2 bytes per struct entry, which would probably give you ~300 bytes more of savings. (Well 300 bytes directly, you'd still have to track it in the for loop, but that should be almost free.

Reply:
@WickedShell yes, I'm planning on doing that as a follow-up PR.

Reply:
https://github.com/ArduPilot/ardupilot/pull/18529